test/e2e/transformations/xmltojson/main.go

```diff
@@ -0,0 +1,173 @@
+)
+
+var _ = Describe("XMLToJSON Transformation", func() {
+	f := framework.New("xmltojsontransformation")
+	var ns string
+	var sink *duckv1.Destination
+	var trnsClient dynamic.ResourceInterface
+	var trans *unstructured.Unstructured
+	var transURL *url.URL
+
+	Context("a Transformation is deployed with K_SINK", func() {
+		BeforeEach(func() {
+			ns = f.UniqueName
+
+			By("creating an event sink", func() {
+				sink = bridges.CreateEventDisplaySink(f.KubeClient, ns)
+				Expect(sink).NotTo(BeNil())
+			})
+
+			By("creating a transformation object", func() {
+				var err error
+				gvr := transAPIVersion.WithResource(transformationResource)
+				trnsClient = f.DynamicClient.Resource(gvr).Namespace(ns)
+				trans, err = createTransformation(trnsClient, ns, "test-xmltojson-", sink)
+				Expect(err).ToNot(HaveOccurred())
+				trans = ducktypes.WaitUntilReady(f.DynamicClient, trans)
+				transURL = ducktypes.Address(trans)
+				Expect(transURL).ToNot(BeNil())
+			})
+
+		})
+		When("a XML payload is sent", func() {
+			BeforeEach(func() {
+				sentEvent := newXMLHelloEvent(f)
+				job := e2ece.RunEventSender(f.KubeClient, ns, transURL.String(), sentEvent)
+				apps.WaitForCompletion(f.KubeClient, job)
+			})
+
+			Specify("should generate a JSON event at the sink", func() {
+				var receivedEvents []cloudevents.Event
+				readReceivedEvents := readReceivedEvents(f.KubeClient, ns, sink.Ref.Name, &receivedEvents)
+
+				const receiveTimeout = 10 * time.Second
+				const pollInterval = 500 * time.Millisecond
+				Eventually(readReceivedEvents, receiveTimeout, pollInterval).ShouldNot(BeEmpty())
+				Expect(receivedEvents).To(HaveLen(1))
+
+				e := receivedEvents[0]
+				Expect(e.Type()).To(Equal("e2e.test"))
+				Expect(string(e.DataEncoded)).To(Equal(expectedResponseEvent))
+			})
+		})
+	})
+
+	Context("a Transformation is deployed without K_SINK", func() {
+		BeforeEach(func() {
+			ns = f.UniqueName
+		})
+		When("the service is created", func() {
+			var err error
+			BeforeEach(func() {
+				gvr := transAPIVersion.WithResource(transformationResource)
+				trnsClient = f.DynamicClient.Resource(gvr).Namespace(ns)
+				trans, err = createTransformation(trnsClient, ns, "test-xmltojsonreplier-", nil)
+				Expect(err).ToNot(HaveOccurred())
+			})
+			Specify("it becomes ready", func() {
+				trans = ducktypes.WaitUntilReady(f.DynamicClient, trans)
+				Expect(trans).ToNot(BeNil())
+			})
+		})
+	})
+})
+
+// newXMLHelloEvent generates a CloudEvent with dummy values and an XML data payload.
+func newXMLHelloEvent(f *framework.Framework) *cloudevents.Event {
+	event := cloudevents.NewEvent()
+	event.SetID("0000")
+	event.SetType("e2e.test")
+	event.SetSource("e2e.triggermesh")
+	event.SetExtension("iotriggermeshe2e", f.UniqueName)
+	if err := event.SetData(cloudevents.ApplicationXML, `<note><to>Tove</to><from>Jani</from><heading>Reminder</heading><body>Dont forget me this weekend</body></note>`); err != nil {
```

Comment:
To my previous comment, the error is right here: a string is passed, but the SDK tries to convert that string to XML instead of accepting it as raw XML data.
Here is the fix: https://go.dev/play/p/Gg5gP15l240

Reply:
got it!

Reply:
Thank you!